auth/postgres/init.go

```diff
@@ -54,9 +54,9 @@ func migrateDB(db *sqlx.DB) error {
 						expires_at  TIMESTAMP,
 						PRIMARY KEY (id, issuer_id)
 					)`,
-					`CREATE extension LTREE`,
+					`CREATE EXTENSION IF NOT EXISTS LTREE`,
 					`CREATE TABLE IF NOT EXISTS group_type (
-						id INTEGER UNIQUE NOT NULL,
+						id SMALLINT UNIQUE NOT NULL,
```

Comment:
Consider using database `ENUM`.

Reply:
doesnt fit here
this doesnt work with enum
```FOREIGN KEY (group_id, type) REFERENCES groups (id, type),```

Reply:
You don't have to use a foreign key later but use a newly created enumerated type (say GroupType or GroupKind). In this particular case, the only foreign key would be `group_id`, while `type` is simply a column of GroupType/GroupKind.